doc: Endo top-level README (#607)
Agoric/SES-shim: packages/ses/README.md

# Secure ECMAScript (SES)

Secure ECMAScript (SES) is an execution environment that provides fine-grained
sandboxing with Compartments.

* **Compartments** Compartments are separate execution contexts: each one has its own global object and global lexical scope.

> * **Frozen realm** Compartments share their intrinsics to avoid identity discontinuity. By freezing the intrinsics, SES removes programs abilities to interfere with each other.

Reviewer: ```suggestion
* **Frozen realm** Compartments share their intrinsics to avoid identity discontinuity. By freezing the intrinsics, SES removes programs' abilities to interfere with each other.
```

Reviewer: or
```suggestion
* **Frozen realm** Compartments share their intrinsics to avoid identity discontinuity. By freezing the intrinsics, SES prevents programs from interfering with each other.
```
or
```suggestion
* **Frozen realm** Compartments share their intrinsics to avoid identity discontinuity. By freezing the intrinsics, SES protects programs from each other.
```

> * **Strict mode** SES enables JavaScript strict mode that enhances security, for example by changing some silent errors into throw errors.

Reviewer: ```suggestion
* **Strict mode** SES enforces JavaScript strict mode that enhances security, for example by changing some silent errors into throw errors.
```

> * **POLA** (Principle of Least Authority) By default, Compartments receive no ambient authorithy. They are created without host-provided APIs, (for example no XMLHttpRequest).

Reviewer: I'm not sure how to rephrase, but the point is that, for example, `XMLHttpRequest` is not provided implicitly. It may still appear in the start compartment, and it may be explicitly endowed to other compartments.

Author: Proposed https://github.com/Agoric/SES-shim/commit/0f59f87bdc0a965b9c2b4ab175db5b0652b273fd

[Learn about the SES specification](https://github.com/tc39/proposal-ses).

[Learn how to use SES in your own project](https://ses-secure-ecmascript.readthedocs.io/en/latest).

Secure ECMAScript (SES) is a frozen environment for running ECMAScript
(Javascript) 'strict' mode programs with no ambient authority in their global
scope, and with the addition of Compartments to evaluate third-party code
safely.
By freezing everything accessible from the global scope, it removes programs'
abilities to interfere with each other, and thus enables isolated evaluation of
arbitrary code.

SES runs atop an ES6-compliant platform, enabling safe interaction of
mutually-suspicious code, using object-capability -style programming.

See https://github.com/Agoric/Jessie to see how SES fits into the various
flavors of confined ECMAScript execution. And visit

> https://rawgit.com/Agoric/ses-shim/master/demo/ for a demo.

Reviewer: This link doesn't work and there is not yet any replacement link.
We now have a working demo again, but no website to visit to see it. Do we have a website ready to go for this?

Author: We can at least temporarily lean on @dckc’s personal Netlify deployment. Do we have a deployment link we can use?

Author: Added ses-demos.netlify.app/demos
